app/src/test/java/tech/relaycorp/gateway/domain/publicsync/CollectParcelsFromGatewayTest.kt

```diff
@@ -107,7 +108,8 @@ class CollectParcelsFromGatewayTest : BaseDataTestCase() {
             .thenReturn(flowOf(parcelCollection, parcelCollection))
         val parcel = mock<Parcel>()
         val parcelAddress = "1234"
-        whenever(parcel.recipientAddress).thenReturn(parcelAddress)
+        val mockRecipient = Recipient("0deadbeef", parcelAddress)
```

Comment:
On line 125, we should be using `mockRecipient.id`, not `parcelAddress`. That's why the regression introduced in the unit under test didn't break this test.

```kotlin
verify(notifyEndpoints, times(2)).notify(eq(MessageAddress.of(mockRecipient.id)))
```

(GitHub doesn't allow me to comment on that line as it's too far from the diff)